### PR TITLE
Add a logger with different log levels

### PR DESCRIPTION
**Pull request by kerrmarin**
Don't really know how/if we should test this 🤷‍♂️

### CODE REVIEW

**Comment by atomoil** [comment]
If you're gonna add a logger, shouldn't it use the modern "unified logging system" - https://developer.apple.com/documentation/os/logging  ?

File: StylableSwiftUI/Classes/Logger.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    public var level: Level
+
+    public enum Level: Int {
+        case debug = 0
+        case info
+        case warning
+        case error
+    }
+
+    public init() {
```
Comment (deanWombourne) [inline StylableSwiftUI/Classes/Logger.swift]: Why is the init method public - do we ever want anyone outside this library creating one?

File: StylableSwiftUI/Classes/Logger.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        #else
+            self.level = .warning
+        #endif
+    }
+
+    private func shouldLog(_ level: Level) -> Bool {
+        return self.level.rawValue <= level.rawValue
+    }
+
+    public func log(_ items: Any..., separator: String = " ", terminator: String = "\n", level: Level = .info) {
```
Comment (deanWombourne) [inline StylableSwiftUI/Classes/Logger.swift]: Do we need this to be public - would we ever want anyone outside this library hijacking our logger?

File: StylableSwiftUI/Classes/Logger.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        guard self.shouldLog(level) else {
+            return
+        }
+
+        let printString = items.map { "\($0)" }.joined(separator: separator)
```
Comment (deanWombourne) [inline StylableSwiftUI/Classes/Logger.swift]: (trivial) I suspect it's a little better performing to do `String(describing: $0)`?

**Review by deanWombourne — APPROVED** [review]

**Review by atomoil — APPROVED** [review]

**Pull request merged**